### DISCUSSION TITLE
Best way to include dependencies hidden inside Rnw and Rmd reports

### DISCUSSION THREAD

**Discussion by a-maldet** [ropensci/targets#614] [Help] [answered]
Hello!

Thanks for the great package! I am currently trying to set up an automated workflow for generating an Rnw based article (spread across many Rnw's). 
Is there a recommended way to include dependencies to other targets manually (without relying on the dependency detection)?
Similar to the `prerequisities` in gnu-make?

For example, I have the situation, that there are some dependencies hidden inside of my Rnw's that are not detected by the dependency detection.
The only solution I could think of is a "dummy call" at the beginning of the code block:

```r
list(
  tar_target(data_for_rnw, iris),
  tar_target(my_rnw, "file.Rnw", format = "file"),
  tar_target(
    my_tex,
    {
      invisible(data_for_rnw)     # Dependency to `data_for_rnw`
      knitr::knit(my_rnw)
    },
    format = "file"
  )
)
```

But maybe, you have a build in solution, I haven't seen?
Thanks and cheers!
Adrian

**Comment by wlandau** [comment, marked as answer]
I wrote `tarchetypes::tar_render()` for this situation: https://docs.ropensci.org/tarchetypes/reference/tar_render.html, https://books.ropensci.org/targets/files.html#literate-programming. Dependencies are detected from calls to `tar_load()` and `tar_read()` in the active R code chunks of the R Markdown source file.

**Reply by a-maldet** [reply]
Wow, you overlooked nothing! Thanks! I was looking in the wrong place. In my usecase it will be `tarchetypes::tar_knit()`, since it is Rnw.
